src/ImageProcessorCore/Numerics/Rational.cs

```diff
@@ -0,0 +1,1143 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Rational.cs" company="James South">
+//   Copyright (c) James South.
+//   Licensed under the Apache License, Version 2.0.
+// </copyright>
+// <summary>
+//   Represents a rational number. Any number that can be expressed as the quotient or fraction p/q of two
+//   numbers, p and q, with the denominator q not equal to zero.
+//   <remarks>
+//   Adapted from <see href="https://github.com/mckamey/exif-utils.net" /> by Stephen McKamey.
+//   </remarks>
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ImageProcessorCore
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Globalization;
+    using System.Reflection;
+
+    /// <summary>
+    /// Represents a rational number. Any number that can be expressed as the quotient or fraction p/q of two 
+    /// numbers, p and q, with the denominator q not equal to zero.
+    /// <remarks>
+    /// Adapted from <see href="https://github.com/mckamey/exif-utils.net"/> by Stephen McKamey.
+    /// </remarks>
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type to assign to the numerator and denominator components.
+    /// </typeparam>
+    [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Reviewed. Suppression is OK here. Better readability.")]
+    public struct Rational<T> :
+        IConvertible,
```

Comment:
@stbraley  Here, give this a shot.

https://gist.github.com/JimBobSquarePants/8e8d9282511c7d6fde1da67d835d584f

`TypeCode` _should_ be easily swappable for `Type` I think in the rest of the code and I've manually implemented the `IConvertable` stuff. The trickiest part to implement is probably getting the `ConstructorInfo` but I think I've cracked it.

All this is untested btw so please give it a thorough look over.

Cheers for picking this up again btw. I appreciate that you are busy and it means a lot to me that you are making the effort. 👍